config-model/src/main/java/com/yahoo/vespa/model/Host.java

```diff
@@ -42,16 +42,14 @@ private Host(AbstractConfigProducer parent, String hostname, boolean runsConfigS
 
     private void checkName(HostSystem parent, String hostname) {
         // Give a warning if the host does not exist
-        // Host exists - warn if given hostname is not a fully qualified one.
-        String canonical = hostname;
         try {
-            canonical = parent.getCanonicalHostname(hostname);
+            Object address = java.net.InetAddress.getByName(hostname);
         } catch (UnknownHostException e) {
-            deployLogger().log(Level.WARNING, "Unable to find canonical hostname of host: " + hostname);
+            deployLogger().log(Level.WARNING, "Unable to lookup IP address of host: " + hostname);
         }
-        if ((null != canonical) && (! hostname.equals(canonical))) {
+        if (! hostname.contains(".")) {
```

Comment:
Should we require that the hostname contains '.' if it already contains the highly inappropriate '_'?

Reply:
since vespa-detect-hostname runs SocketAddress::normalize() it's highly expected that hostname will get a domain suffix, even when it contains "_" or is otherwise abnormal.

Reply:
I would expect symbolic container names with _ to be considered canonical by the docker dns proxy. I have not tested this, so I do not know.